app/plugin_install.go

```diff
@@ -308,6 +309,11 @@ func (a *App) RemovePlugin(id string) *model.AppError {
 }
 
 func (a *App) removePlugin(id string) *model.AppError {
+	pluginEnabled := false
+	if pluginState, ok := a.Config().PluginSettings.PluginStates[id]; ok && pluginState.Enable {
+		pluginEnabled = true
+	}
```

Comment:
Are these changes related to the canary plugins? Not sure if this change was from testing something else

Reply:
When we remove plugin with the canary tag it will be good to fallback on the same plugin without canary tag(if there is one). Here I save if removed plugin is enabled or not to enable or disable newly installed plugin. 
Actually I'm 0/5 if we should have this behavior or not.

Reply:
I suggest we remove. It would have no effect because it always disables the plugin before uninstallation [here](https://github.com/mattermost/mattermost-server/blob/3731724a65dffdc3c05b48ed9b3d974369cb47f7/app/plugin_install.go#L319).

Reply:
Exactly it disables before removing that is why before disabling we save the state, remove the plugin, install plugin with no canary tag and after that enable the state.
Am I missing something? Why will it have no effect?

Reply:
I was the one missing that it was passed in to the `installPluginWithNoCanaryTag` where you enable it again. Got it!